sections/production/assigntransactionid.md

```diff
@@ -8,7 +8,65 @@ A typical log is a warehouse of entries from all components and requests. Upon d
 
 <br/><br/>
 
-### Code example: typical Express configuration
+### Code example: sharing TransactionId among current request functions 
+using [cls-rtracer](https://www.npmjs.com/package/cls-rtracer) (a library based on [async-local-storage](https://nodejs.org/api/async_hooks.html#async_hooks_class_asynclocalstorage), implemented for Express & Koa middlewares and Fastify & Hapi plugins)
+
+```javascript
+const express = require('express');
+const rTracer = require('cls-rtracer');
+
+const app = express();
+
+app.use(rTracer.expressMiddleware());
+
+app.get('/getUserData/{id}', async (req, res, next) => {
+    try {
+        const user = async usersRepo.find(req.params.id);
+
+        // At any point in the app after cls-rtracer middleware was initialized, even when 'req' object doesn't exist, the TransactionId is reachable
+        const transactionId = rTracer.id();
+        logger.info(`user ${user.id} data was fetched successfully`, { transactionId });
+
+        res.json(user);
+    } catch (err) {
+        // If not for the example, I'd recommand using rTracer.id() from inside the logger component, to prevent from using it all over the code
+        logger.error(`error while fetching user id ${req.params.id} data`, { transactionId: rTracer.id(), error: err });
+
+        next(err);
+    }
+})
+```
+<br/><br/>
+
+### Code example: sharing TransactionId among microservices 
+
+```javascript
+// cls-tracer has the ability to store the TransactionId on your service outgoing requests headers, and extract the TransactionId from incoming requests headers, just by overriding the default middleware config
+app.use(rTracer.expressMiddleware({
+    // Add TransactionId to the header
+    echoHeader: true,
+    // Respect TransactionId from header
+    useHeader: true,
+    // TransactionId header name
+    headerName: 'x-transaction-id'
+}));
+
+const axios = require("axios");
+
+// Now, the external service will automaticlly get the current TransactionId as header
+const response = await axios.get('https://externalService.com/api/getAllUsers');
+```
+<br/><br/>
+
+
+There are two restrictions on using cls-rtracer, due to its dependence on async-local-storage (You can think of async-local-storage as the node.js alternative to thread local storage, more details [here](https://www.freecodecamp.org/news/async-local-storage-nodejs/)):
+1. async-local-storage requires Node v.14. 
+2. async-local-storage is based on a lower level construct in Node called async_hooks which is still experimental, so you may have the fear of performance problems. Even if they do exist, they are very negligible, but you should make your own considerations.

```

Comment:
I think those points should be kept as they apply to the underlying `async-local-storage` even if we remove `cls-rtracer` form the main example.